### ISSUE TITLE
Question of @type-ddd v4 : where is CustomNumberValueObject, CustomStringValueObject and DateValueObject ?

### ISSUE THREAD

**Issue by GaetanCottrez** [4lessandrodev/type-ddd#446]
I noticed that in v4, `CustomNumberValueObject`, `CustomStringValueObject`, and `DateValueObject` no longer exist. 

Have they been removed in v4, or is this a regression?

**Comment by 4lessandrodev** [comment]
You are correct. The `DateValueObject` was indeed an oversight on my part in version 4 and will be reintroduced in a future update. As for the `CustomNumberValueObject` and `CustomStringValueObject`, they were intentionally removed. I realized that it's better to use specific value objects rather than generic ones, as this promotes more precise domain modeling and clearer code.

**Comment by 4lessandrodev** [comment]
I just found the package published at [https://www.npmjs.com/package/@type-ddd/date](https://www.npmjs.com/package/@type-ddd/date) and available in `/packages/date`.

**Comment by GaetanCottrez** [comment]
Okay, I understand the approach and I support it. 

However, it will be important to communicate well and explain how to migrate from v3 to v4 for these value objects, as some users of the library might get frustrated. 

This could also be a barrier to migrating to v4 since the Value Objects were not marked as deprecated. ;-)

I close this issue.
